core/primitives/src/block.rs

```diff
@@ -591,15 +591,22 @@ impl Block {
         }
     }
 
-    pub fn shards_congestion_info(&self) -> HashMap<ShardId, CongestionInfo> {
-        self.chunks()
-            .iter()
-            .enumerate()
+    pub fn shards_congestion_info(&self) -> HashMap<ShardId, ExtendedCongestionInfo> {
+        let mut result = HashMap::new();
+
+        for chunk in self.chunks().iter() {
+            let shard_id = chunk.shard_id();
             // TODO(congestion_control): default is not always appropriate!
-            .map(|(i, chunk_header)| {
-                (i as ShardId, chunk_header.congestion_info().unwrap_or_default())
-            })
-            .collect()
+            let congestion_info = chunk.congestion_info().unwrap_or_default();
+            let height_included = chunk.height_included();
+            let height_current = self.header().height();
+            let missed_chunks_count = height_current - height_included;
```

Comment:
nit: consider using `checked_sub` with something like `.expect("current cannot be older than included")` to make it more obvious that the code technically has a panic path here but that we have semantic guarantees that make it unreachable.